core/core.mk

```diff
@@ -11,7 +11,7 @@ include mk/config.mk
 include core/arch/$(ARCH)/$(ARCH).mk
 
 PLATFORM_$(PLATFORM) := y
-PLATFORM_FLAVOR_$(PLATFORM_FLAVOR) := y
+PLATFORM_FLAVOR_$(subst -,_,$(PLATFORM_FLAVOR)):= y
```

Comment:
I'm not sure about this patch. As it is, it is a bit incomplete: why allow hyphens in the flavor but not in the platform? Also, the line that splits a composite PLATFORM into PLATFORM and PLATFORM_FLAVOR could be modified to address the limitation you're mentioning in the commit message (split at the first hyphen by default).

However it may be more reasonable to stick to the "no hyphen" practice. We have two instances already, `qemu_virt` and `salvator_h3`.

Reply:
> why allow hyphens in the flavor but not in the platform?

With legacy build scripts, an hyphen in `PLATFORM` is implicitly a separation between a platform and its flavor. Do you want to discard this? And expect all build explicitly set `PLATFORM_FLAVOR` when needed to specify a flavor?

> However it may be more reasonable to stick to the "no hyphen" practice. 

Why not allowing `PLATFORM_FLAVOR` to contain hyphen(s)?

Reply:
> With legacy build scripts, an hyphen in PLATFORM is implicitly a separation between a platform and its flavor. Do you want to discard this?

No, I would make it "the **first** hyphen is a separation between (base) platform and its flavor".

> Why not allowing PLATFORM_FLAVOR to contain hyphen(s)?

Well, it is a bit weird to allow hyphens in flavor but not in the base platform, and it makes things a bit complicated. The hyphen cannot stay in the derived name $(PLATFORM_$(PLATFORM_FLAVOR)), so it is simpler to forbid the hyphen everywhere.

Anyway, if you prefer to allow hyphens in the flavor, it's OK, but I think this is the way to do it:

```diff
diff --git a/Makefile b/Makefile
index 7caab01bb..e2578017a 100644
--- a/Makefile
+++ b/Makefile
@@ -30,11 +30,20 @@ include $(CFG_OPTEE_CONFIG)
 endif
 
 # If $(PLATFORM) is defined and contains a hyphen, parse it as
-# $(PLATFORM)-$(PLATFORM_FLAVOR) for convenience
+# $(PLATFORM)-$(PLATFORM_FLAVOR) for convenience (split at first hyphen)
 ifneq (,$(findstring -,$(PLATFORM)))
-ops := $(join PLATFORM PLATFORM_FLAVOR,$(addprefix =,$(subst -, ,$(PLATFORM))))
-$(foreach op,$(ops),$(eval override $(op)))
+_P := $(firstword $(subst -, ,$(PLATFORM)))
+_F := $(subst -,_,$(subst $(_P)-,,$(PLATFORM)))
+ifneq (,$(_F))
+ifneq (,$(PLATFORM_FLAVOR))
+$(error PLATFORM_FLAVOR is set but PLATFORM contains -<flavor>)
+endif
+endif
+$(eval override PLATFORM := $(_P))
+$(eval override PLATFORM_FLAVOR := $(_F))
 endif
+# Any remaining hyphen is converted to an underscore
+override PLATFORM_FLAVOR := $(subst -,_,$(PLATFORM_FLAVOR))
 
 # Make these default for now
 $(call force,ARCH,arm)
```

The differences with your proposal:
- `make stm32mp1-stm32mp157c-ev1` works as expected
- Hyphens are always replaced in the flavor name, not just in the C pre-processor macro $(PLATFORM_$(PLATFORM_FLAVOR)). It's just more consistent.
- `make PLATFORM=foo-bar PLATFORM_FLAVOR=<something>` will report an error because flavor is given twice.

Reply:
Please keep it simple and skip hyphens in flavors.

Reply:
Ok, let's keep it simple.
I will discard my change.